cmake/LinkLibrary.cmake

```diff
@@ -0,0 +1,22 @@
+function(link_library NAME)
+    set(args ${ARGN})
+
+    # Check if any of the argument is ENCOURAGED
+    if ("ENCOURAGED" IN_LIST args)
+        set(ENCOURAGED YES)
+    else ("ENCOURAGED" IN_LIST args)
+        set(ENCOURAGED NO)
+    endif ("ENCOURAGED" IN_LIST args)
+
+    if (${NAME}_FOUND)
+        string(TOUPPER "${NAME}" UCNAME)
+        add_definitions(-DWITH_${UCNAME})
+        include_directories(${${NAME}_INCLUDE_DIRS})
+        target_link_libraries(openttd ${${NAME}_LIBRARIES})
```

Comment:
I would allow to pass a target to this function. If a target is given then just link that, and only otherwise use ```include_directories()``` and link to the *_LIBRARIES variable. Targets can transport additional things like link flags, compile definitions and so on and should generally be preferred.

Reply:
@DerDakon I agree and addressed this in https://github.com/TrueBrain/OpenTTD/pull/11